code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -882,29 +882,43 @@ obj/item/assembly/radio_horn/receive_signal()
 	var/thingsneeded = null
 	var/obj/item/ammo/bullets/result = null
 	var/obj/item/accepteditem = null
+	var/craftname = null
+	var/obj/item/material = null
 
 	proc/craftwith(obj/item/craftingitem, obj/item/frame, mob/user)
+
 		if (istype(craftingitem, accepteditem)) //success! items match
 			src.thingsneeded --
+			if (craftingitem.amount > 1)//it's a little weird but it's one of the lesser messy ways
+				material = craftingitem.split_stack(1)
+			else
+				material = craftingitem
```

Comment:
Rather than doing this. Just set `material = craftingitem`, then at line 906 leverage `change_stack_amount(...)`:
https://github.com/goonstation/goonstation/blob/17d2fc3f2b4b26385f9338fa89e05576fc792cdb/code/obj/item.dm#L547

This should handle the qdel for you.

Reply:
The reservation I have with these two suggestions is that I'm unsure with how they'll interact with non-material items, but I'll work on them

Reply:
The variable `amount` and proc `change_stack_amount` are both from the base `obj/item` object. Meaning any item will inherit them. So as long as you're type checking whatever is being inserted is an item, it should be fine.